lib/elixir_analyzer/exercise_test/common_checks/exemplar_comparaison.ex

```diff
@@ -0,0 +1,27 @@
+defmodule ElixirAnalyzer.ExerciseTest.CommonChecks.ExemplarComparaison do
+  @moduledoc """
+  Compares the solution to the exemplar solution for concept exercises.
+  Ignores practice exercises.
+  """
+
+  alias ElixirAnalyzer.Constants
+  alias ElixirAnalyzer.Comment
+
+  @spec run(Macro.t(), nil | Macro.t()) :: [{:pass | :fail | :skip, %Comment{}}]
+  def run(_ast, nil), do: []
+
+  def run(code_ast, exemplar_ast) do
+    if Macro.to_string(code_ast) == Macro.to_string(exemplar_ast) do
```

Comment:
Both get turned into ASTs first, so differences like single-line `if` vs multiline `if` won't matter, right? That's still less strict than doing a strict string equality on the original sources :)

Btw. is it necessary to turn them back into strings for the comparison?

Reply:
Yes, I agree it is not the absolute strictest check, but it's still pretty strict, all variable names need to be the same, same typespecs and docs (those are not too much of an issue for concept exercise, I know), etc.

It's not strictly speaking necessary to compare strings, at first I was going through both ASTs and neglecting the metadata, but I kept forgetting corner cases... For debugging I was printing the code as strings and I realized I might as well do that. It's one function rather than 25+ lines of code.